### ISSUE TITLE
The content entered by the Chinese input method is displayed repeatedly

### ISSUE THREAD

**Issue by Hhpon** [ianstormtaylor/slate#5023]
**Description**
The compositionEnd event and the beforeInput event trigger the insertText event repeatedly, resulting in repeated typing。

**Recording**
https://user-images.githubusercontent.com/33001067/172849978-2e272d4b-ff1a-43b7-966e-68fc9508b062.mov

**Sandbox**
[A link to a sandbox where the error can be reproduced. (You can start from the base sandbox here: https://codesandbox.io/s/slate-reproductions-c7gyg or refer to the Slate website too.)](https://codesandbox.io/s/slate-editor-demo-7yoh58?file=/src/App.js)

**Steps**
To reproduce the behavior:
1. open Safari Broswer
2. Change userAgent to "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko)"
3. Switch Chinese input method, type anything

**Expectation**
I know this problem is caused by duplication of insert text with onCompositionEnd event and onBeforeInput event. Shouldn't apple webkit webview be drained in onCompositionEnd event；

**Environment**
- Slate Version: 0.80.0
- Operating System: macos
- Browser: macos webview  userAgent Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko)

**Context**
[same type issue](https://github.com/ianstormtaylor/slate/pull/4702)

**Comment by RyuzakiZ99** [comment]
Has this bug already been fixed? Or perhaps some sort of workaround has been found?

**Comment by manly0305** [comment]
Same issue

**Comment by Hhpon** [comment]
> Has this bug already been fixed? Or perhaps some sort of workaround has been found?
Is your current slate operating environment the same?“macos webview userAgent Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko)”

**Comment by RyuzakiZ99** [comment]
@Hhpon Yes, same environment. I'm trying to set a custom user-agent for the browser but slate doesn't seem to recognize it

`
useEffect(() => {
    let userAgent = window.navigator.userAgent
    console.info(userAgent)
    Object.defineProperty(navigator, 'userAgent', {
      get: function () {
        return 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15'
      }
    })
    userAgent = window.navigator.userAgent
    console.info(userAgent)
  }, [])
`

Is there a way to set it so that Slate picks up? The code above seems to log the new user-agente correctly, but is still doesn't work.

**Comment by Hhpon** [comment]
> @Hhpon Yes, same environment. I'm trying to set a custom user-agent for the browser but slate doesn't seem to recognize it
> 
> `useEffect(() => { let userAgent = window.navigator.userAgent console.info(userAgent) Object.defineProperty(navigator, 'userAgent', { get: function () { return 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Safari/605.1.15' } }) userAgent = window.navigator.userAgent console.info(userAgent) }, [])`
> 
> Is there a way to set it so that Slate picks up? The code above seems to log the new user-agent correctly, but is still doesn't work.

hello，Is your slate running in a macos webview? My solution at the time was uesrAgent modified by the developer of webview.

**Comment by RyuzakiZ99** [comment]
I'll try to change the webview, thank you!
Also noticed that the bug also happens with a browser with user-agent:
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) MobileSafari/18614.3.7.1.5
